src/lib.rs

```diff
@@ -9,24 +9,27 @@ use curv::elliptic::curves::traits::{ECPoint, ECScalar};
 use curv::BigInt;
 use multi_party_ecdsa::protocols::multi_party_ecdsa::gg_2020::state_machine::keygen::LocalKey;
 use paillier::{
-    Add, Decrypt, Encrypt, EncryptWithChosenRandomness, Mul, Paillier, Randomness, RawCiphertext,
-    RawPlaintext,
+    Add, Decrypt, DecryptionKey, Encrypt, EncryptWithChosenRandomness, EncryptionKey,
+    KeyGeneration, Mul, Paillier, Randomness, RawCiphertext, RawPlaintext,
 };
 use std::fmt::Debug;
 use zeroize::Zeroize;
+use zk_paillier::zkproofs::{NICorrectKeyProof, SALT_STRING};
 
 // Everything here can be broadcasted
-#[derive(Clone, PartialEq)]
+#[derive(Clone)]
 pub struct RefreshMessage<P> {
     party_index: usize,
     fairness_proof_vec: Vec<FairnessProof<P>>,
     coefficients_committed_vec: VerifiableSS<P>,
     points_committed_vec: Vec<P>,
     points_encrypted_vec: Vec<BigInt>,
+    ek: EncryptionKey,
+    dk_correctness_proof: NICorrectKeyProof,
 }
 
 impl<P> RefreshMessage<P> {
-    pub fn distribute(old_key: &LocalKey<P>) -> Self
+    pub fn distribute(old_key: &LocalKey<P>) -> (Self, DecryptionKey)
```

Comment:
why not overwrite the existing `dk` in LocalKey?

Reply:
Makes a ton of sense.

Reply:
The naming `old_key` made me think that a new one will be generated at some point. Maybe we would like to rename that only to local_key if we want to reuse it.

Reply:
good idea!